### PR TITLE
WT-12641 Create a new stat for the number of bytes estimated to be moved by compact

### PR DESCRIPTION
**Pull request by Sean04**

### CODE REVIEW

File: src/block/block_compact.c [hunk 1]
```diff
@@ -89,6 +90,8 @@ __wt_block_compact_get_progress_stats(
     WT_STAT_DATA_SET(session, btree_compact_pages_rewritten, block->compact_pages_rewritten);
     WT_STAT_DATA_SET(
       session, btree_compact_pages_rewritten_expected, block->compact_pages_rewritten_expected);
+    WT_STAT_DATA_SET(
```
Comment (etienneptl) [inline src/block/block_compact.c:93]: Can we A->Z this list please?

File: src/block/block_compact.c [hunk 2]
```diff
@@ -403,12 +406,13 @@ __block_compact_estimate_remaining_work(WT_SESSION_IMPL *session, WT_BLOCK *bloc
...
     block->compact_pages_rewritten_expected = block->compact_pages_rewritten + total_pages_to_move;
+    block->compact_bytes_rewritten_expected =
+      block->compact_pages_rewritten_expected * (uint64_t)avg_block_size;
 
     __wt_verbose_level(session, WT_VERB_COMPACT, verbose_level,
       "%s: expecting to move approx. %" PRIu64 " more pages (%" PRIu64 "MB), %" PRIu64
       " total, target %" PRIu64 "MB (%" PRIu64 "B)",
-      block->name, total_pages_to_move,
-      total_pages_to_move * (uint64_t)avg_block_size / WT_MEGABYTE,
+      block->name, total_pages_to_move, block->compact_bytes_rewritten_expected / WT_MEGABYTE,
```
Comment (etienneptl) [inline src/block/block_compact.c:415]: The code looks different now. Before it used `total_pages_to_move` which does not take into account the previous `block->compact_pages_rewritten`.

Comment (Sean04) [reply src/block/block_compact.c:415]: Ah that's true, I'll change this back since the message is about how _more_ bytes are moved.

Comment (Sean04) [reply src/block/block_compact.c]: https://github.com/wiredtiger/wiredtiger/pull/10354/commits/5f037aa5a80b6534e7c091ad925e7aa4f79dbe39

File: src/block/block_compact.c (hunk 1 above)
Comment (Sean04) [reply src/block/block_compact.c]: https://github.com/wiredtiger/wiredtiger/pull/10354/commits/bcb2bf27f40105d04ff087f9ccea66d81ee108cf

**Review by etienneptl — APPROVED** [review]
LGTM!

**Pull request merged**